docs/decisions/0009-persistence-and-retries-for-events.rst

```diff
@@ -0,0 +1,25 @@
+Persistence and retries for events
+==================================
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+`event-routing-backends` transmits events to configured Learning Record Consumer (LRC) via http protocol in near real-time. A strategy needs to be implemented to handle the case when an LRC's link is down.
+
+Decision
+--------
+
+1. Retry attempts shall be made for all events for a configured limit of time and number of retries.
+
+2. A limited type of events (lets call them persisted events) shall be stored in a database after all retry attempts have been exhausted. List of persisted events shall contain events that an LRC may use for record keeping such as course enrolment and completion events.
```

Comment:
[Nit] `enrolment` -> `enrollment`

Reply:
Add clarification that the persisted task (via `celery-utils`) will include just-enough information about the event so the event could get resent appropriately after persistence.

Reply:
[naming] @ziafazal suggested naming these types of events as "business critical".

[request] Update the Event-listing Docs to specific which events we are considering as "business critical".

Reply:
1. Done
2. Done
3. Done
4. Link will be added after updating the docs